modules/calib3d/src/usac/degeneracy.cpp

```diff
@@ -67,16 +68,17 @@ Ptr<EpipolarGeometryDegeneracy> EpipolarGeometryDegeneracy::create (const Mat &p
 
 class HomographyDegeneracyImpl : public HomographyDegeneracy {
 private:
-    const Mat * points_mat;
-    const float * const points;
+    Mat points_mat;
     const float TOLERANCE = 2 * FLT_EPSILON; // 2 from area of triangle
 public:
     explicit HomographyDegeneracyImpl (const Mat &points_) :
-            points_mat(&points_), points ((float *)points_mat->data) {}
+            points_mat(points_)
+    {}
 
     inline bool isSampleGood (const std::vector<int> &sample) const override {
         const int smpl1 = 4*sample[0], smpl2 = 4*sample[1], smpl3 = 4*sample[2], smpl4 = 4*sample[3];
         // planar correspondences must lie on the same side of any line from two points in sample
+        const float * points = (const float*)points_mat.data;
```

Comment:
Maybe `points_mat.ptr<float>()` would be better? Here and in other similar places.

Reply:
Done!